js/src/forum/components/FlagPostModal.js

```diff
@@ -54,12 +60,24 @@ export default class FlagPostModal extends Modal {
                 {app.translator.trans('flarum-flags.forum.flag_post.reason_inappropriate_text', {
                   a: guidelinesUrl ? <a href={guidelinesUrl} target="_blank"/> : undefined
                 })}
+                {this.reason() === 'inappropriate' ? (
+                  <div>
+                    <p><em>{app.translator.trans('flarum-flags.forum.flag_post.optional_details')}</em></p>
+                    <textarea className="FormControl" value={this.reasonDetail()} oninput={m.withAttr('value', this.reasonDetail)}></textarea>
+                  </div>
+                ) : ''}
               </label>
 
               <label className="checkbox">
                 <input type="radio" name="reason" checked={this.reason() === 'spam'} value="spam" onclick={m.withAttr('value', this.reason)}/>
                 <strong>{app.translator.trans('flarum-flags.forum.flag_post.reason_spam_label')}</strong>
                 {app.translator.trans('flarum-flags.forum.flag_post.reason_spam_text')}
+                {this.reason() === 'spam' ? (
+                  <div>
+                    <p><em>{app.translator.trans('flarum-flags.forum.flag_post.optional_details')}</em></p>
+                    <textarea className="FormControl" value={this.reasonDetail()} oninput={m.withAttr('value', this.reasonDetail)}></textarea>
```

Comment:
When you enter text here, and then choose another category - will the previously entered text appear in that category's text field?

If so, do we want that? :thinking:

Reply:
This is intentional. Our thoughts behind it are to save a user from re-typing their explanation if they accidentaly selected the wrong reason.

Open to thoughts and suggestions here...

Reply:
Okay, just wanted to ensure it's intentional. If it feels good when using it, we can try this out...